implement shaders rewrites and dx9 shader compilation script

Berthalamew opened the pull request:
- "build dx9 shaders.bat" compiles the .fx files and populates the bytecode into headers
- include xxd and sed from w64devkit
- rewrite weather, fog, and particle shaders for fixed effects
- fix snow size depending on resolution
- include d3d9 debug info in debug builds

![00314](https://github.com/user-attachments/assets/e400ce95-bf19-49c7-95c3-5c392bd31f71)

Berthalamew committed:
add function to get local player count

Berthalamew committed:
remove unneeded #includes

Berthalamew committed:
misc changes

Berthalamew committed:
implement shaders rewrites and dx9 shader compilation script

- "build dx9 shaders.bat" compiles the .fx files and populates the bytecode into headers
- include xxd and sed from w64devkit
- rewrite weather, fog, and particle shaders
- include d3d9 debug info in debug builds

nukeulater commented on xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fog.cpp:
```diff
@@ -1,8 +1,309 @@
+	return;
+}
+
+bool __cdecl rasterizer_dx9_draw_atmospheric_fog(int32 a1)
```
this sets up the pipeline, doesn't render anything at all

Berthalamew replied:
function name changed to `rasterizer_dx9_atmospheric_fog_pipeline_setup`

nukeulater commented on xlive/Blam/Engine/render/weather_definitions.h:
```diff
@@ -1,10 +1,10 @@
 #include "tag_files/tag_reference.h"
 
-#define MAX_PARTICLES_LITE = 4096
 #define MAX_WIND_PRIMITIVES = 128
```
= 128?

Berthalamew replied:
fixed

Berthalamew committed:
add suggested changes

- remove unneeded patch for atmospheric fog function
- add enums for other output types
- add asserts in saved_games_async_helper_get_saved_game_bin_path

nukeulater commented on xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_fullscreen_passes.cpp:
```diff
@@ -11,30 +11,92 @@
+    return;
+}
 
+void rasterizer_dx9_fullscreen_calculate_texcoords(const real_rectangle2d* bounds, const real_point2d* location, real_point2d* output)
```
could probably make the texture index a param here

Berthalamew replied:
added parameter with default of 0

nukeulater commented on xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_main.cpp:
```diff
@@ -99,10 +107,14 @@ IDirect3DPixelShader9** local_pixel_shaders_get(void)
     // debugging, get vertex decls currently set
-    //PatchCall(Memory::GetAddress(0x2220CA), DrawPrimitiveUP_hook_get_vertex_decl);
-    //PatchCall(Memory::GetAddress(0x27D746), DrawPrimitiveUP_hook_get_vertex_decl);
+    PatchCall(Memory::GetAddress(0x2220CA), DrawPrimitiveUP_hook_get_vertex_decl);
```
shouldn't be enabled

Berthalamew replied:
removed

Berthalamew committed:
add suggested changes

- fix warnings in dynavobgeom code
- use ustrncpy instead of csmemcpy in game_direct_connect_to_session
- fix fwprintf parameter for ragdoll count
